react/features/breakout-rooms/actions.js

```diff
@@ -0,0 +1,184 @@
+ */
+export function closeRoom(roomId: string, newRoomId?: string) {
+    return (dispatch: Dispatch<any>, getState: Function) => {
+        const { rooms = {} } = getBreakoutRooms(getState);
+        const { participants = {} } = rooms[roomId];
+        const _newRoomId = newRoomId || getMainRoomId(getState);
+
+        Object.keys(participants).forEach(participantId =>
+            dispatch(sendParticipantToRoom(participantId, _newRoomId))
+        );
+    };
+}
+
+/**
+ * Action to remove a breakout room.
+ *
+ * @param {string} breakoutRoomJid - The jid of the breakout room to remove.
+ * @returns {Function}
+ */
+export function removeBreakoutRoom(breakoutRoomJid: string) {
+    return (dispatch: Dispatch<any>, getState: Function) => {
+        const message = {
+            type: JSON_TYPE_REMOVE_BREAKOUT_ROOM,
+            breakoutRoomJid
+        };
+
+        getCurrentConference(getState).sendMessage(message, 'focus');
+    };
+}
+
+
+/**
+ * Action to auto-assign the participants to breakout rooms.
+ *
+ * @returns {Function}
+ */
+export function autoAssignToBreakoutRooms() {
+    return (dispatch: Dispatch<any>, getState: Function) => {
+        const { rooms = {} } = getBreakoutRooms(getState);
+        const breakoutRooms = _.filter(rooms, (room: Object) => !room.isMainRoom);
+
+        if (breakoutRooms) {
+            const participantIds = Array.from(getRemoteParticipants(getState).keys());
+            const length = Math.ceil(participantIds.length / breakoutRooms.length);
+
+            _.chunk(_.shuffle(participantIds), length).forEach((group, index) =>
+                group.forEach(participantId =>
+                    dispatch(sendParticipantToRoom(participantId, breakoutRooms[index].id))
+                )
+            );
+        }
+    };
+}
+
+/**
+ * Action to send a participant to a room.
+ *
+ * @param {string} participantId - The participant id.
+ * @param {string} roomId - The room id.
+ * @returns {Function}
+ */
+export function sendParticipantToRoom(participantId: string, roomId: string) {
+    return (dispatch: Dispatch<any>, getState: Function) => {
+        const conferenceOptions = getConferenceOptions(getState);
+        const id = participantId.indexOf('@') >= 0
+            ? participantId
+            : `${getCurrentRoomId(getState)}@${isInBreakoutRoom(getState)
+                ? `breakout.${conferenceOptions.hosts.domain}`
+                : conferenceOptions.hosts.muc
+            }/${participantId}`;
+        const message = {
+            type: JSON_TYPE_MOVE_TO_ROOM_REQUEST,
+            roomId
+        };
+
+        getCurrentConference(getState).sendMessage(message, id);
+    };
+}
+
+/**
+ * Action to move to a room.
+ *
+ * @param {string} roomId - The room id to move to. If omitted move to the main room.
+ * @returns {Function}
+ */
+export function moveToRoom(roomId?: string) {
+    return (dispatch: Dispatch<any>, getState: Function) => {
+        // FIXME: Move this code to conference.js
```

Comment:
conference.js is legacy, no new code should be added there if possible, so it's ok for it to be here.

Reply:
Thanks for mentioning this. I will remove the comment.